kobotoolbox/kpi
# feat(accessLogExport)!: create new AccessLogExportTask to generate a csv of access logs TASK-871

### 👀 Preview steps

1. Ensure that the `Export all data` button for `ProjectViews` still works as normal and that the email header has the appropriate title: "Project View Report Complete"
2. Ensure that exports for submission data works correctly:
- make submissions to a project
- navigate to the _Project < Data < Downloads_ 
- 🟢 ensure the `Export` button is exporting data correctly
3. Preview the AccessLogExportTask from the shell:
- ℹ️ have a user 
- enter the Django shell: `./manage.py shell_plus`
- get your user: `test_user = User.objects.get(username='test')`
- create an AccessLogExportTask: 
``` 
task = AccessLogExportTask.objects.create(
           user=test_user,
           get_all_logs=False,
           data={'type': 'access_logs_export'},
  )
```
- 🟢 print the task object and notice that an AccessLogExportTask object has been created: `print(task)`
- run the task: `task.run()`
- 🟢 print the result and notice that it generated a link to the csv export: `print(task.result)`

### 💭 Notes
- Changed the field named `user` to `user_url` because naming it  `user` conflicts with an existing field in the `AccessLog` model
- Renamed the file titled `project_view_exports.py` to `data_exports.py` because it now generates exports for both project view and access log exports
- Refactored the `ProjectViewExportTask` to have a base `CommonExportTask` which both the project view and access log export classes inherit from 
- Updated the email (in `kpi/tasks.py`) to have the correct subject title based on the type of export report it is sending
- Alphabetized the functions in `data_exports.py`
